sc-tools/sc-builder/src/scs_translator.cpp

```diff
@@ -114,9 +114,19 @@ bool SCsTranslator::TranslateImpl(Params const & params)
 
   SCsHelper scs(m_ctx, std::make_shared<impl::FileProvider>(params.m_fileName));
   
-  if (!scs.GenerateBySCsText(data))
+  if(params.m_resultStructureUpload)
   {
-    SC_THROW_EXCEPTION(utils::ExceptionParseError, scs.GetLastError());
+      if (!scs.GenerateBySCsText(data, params.m_outputStructure))
```

Comment:
I think we can leave only this "if". We can call scs.GenerateBySCsText(data, params.m_outputStructure), this method will ignore second param if it's not valid